src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -2572,7 +2572,6 @@ Compiler::fgWalkResult Compiler::fgStress64RsltMulCB(GenTree** pTree, fgWalkData
 
     // To ensure optNarrowTree() doesn't fold back to the original tree.
     tree->AsOp()->gtOp1 = pComp->gtNewCastNode(TYP_LONG, tree->AsOp()->gtOp1, false, TYP_LONG);
```

Comment:
`optNarrowTree` might "fold that back" now. You could use a `COMMA` with a volatile store to an unused local to prevent it from doing so, perhaps.

Reply:
Alternatively a debug-only flag might be a bit simpler, since this is just for stress-only code anyway.